### ISSUE TITLE
Create API that other mods can use

### ISSUE THREAD

**Issue by Black-DeathPE** [zhuowei/MCPELauncher#467]
I think it would be really cool if you could create ModPE APIs using one mod (while it is running) and then other mods can use that API. So, for example I could make ModPE.killPlayer(); using one mod (unsure how you could create the API within) and then when it's called in another mod it does the code defined in the first mod.

**Comment by sero583** [comment]
It is already possible, take a look at @Desno365 Minecraft PE Button library.

**Comment by sero583** [comment]
Please close the issue @Black-DeathPE

**Comment by sliceofcode** [comment]
No, that is just copy-pasting code. A true import system would have methods like `require(...);`.

**Comment by Black-DeathPE** [comment]
Exactly. I would like to see something added that by having one script on an API can be used in another script without pasting hooks into one code.

**Comment by zedman2000** [comment]
you can already do this.  I'll post you can example later.

**Comment by Black-DeathPE** [comment]
You can't. Not by having the API in one script and using the API in another.

**Comment by zedman2000** [comment]
Yes, I already do this in one of my mods.

**Comment by zedman2000** [comment]
yes, as I said, I do this in my one of my mods now.
net.zhuoweizhang.mcpelauncher.ScriptManager.callScriptMethod("function2Call",param1,param2,param3,etc);

**Comment by Black-DeathPE** [comment]
Interesting. I'll give this a try. However I'm sure you're not supposed to do this but the java bridge allows you to :P

**Comment by RandomAltThingy** [comment]
It should be so like

```javascript
//Or whatever
function newLevel(){
   ModPE.callFunction(String func);
}
```

**Comment by Black-DeathPE** [comment]
I think that's a good idea, however you could have it as this:

function sayHello() {
  clientMessage("Hello!");
}

ModPE.defineAPI("sayHello", sayHello());

it could then take parameters as the hook does. Idk how namespace could be used

**Comment by RandomAltThingy** [comment]
@Black-DeathPE Nah, make it like <code>ModPE.defineMethod(scriptName, methodName);</code>

Gotta have the script name there in case 2 scripts have the same method name

**Comment by Black-DeathPE** [comment]
By using namespaces this could be bypassed, so for functions in my mod I could have:

EntityManager.killZombies();
